File: 3536.maximum-product-of-two-digits.py
#
# @lc app=leetcode id=3536 lang=python
#
# [3536] Maximum Product of Two Digits
#

# @lc code=start
class Solution(object):
    def maxProduct(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n < 100:
            return (n%10) * (n//10)
        l = list(str(n))
        l.sort()
        return int(l[-1]) * int(l[-2])
        
# @lc code=end

File: 3536.maximum-product-of-two-digits.java
/*
 * @lc app=leetcode id=3536 lang=java
 *
 * [3536] Maximum Product of Two Digits
 */

// @lc code=start
class Solution {
    public int maxProduct(int n) {
        char[] chars = String.valueOf(n).toCharArray();
        Arrays.sort(chars);
        int one = chars[chars.length - 1] - '0';
        int two = chars[chars.length - 2] - '0';
        return one * two;
    }
}
// @lc code=end

